### ISSUE TITLE
Cannot get html lightning language support for aura

### ISSUE THREAD

**Issue by dansad** [forcedotcom/salesforcedx-vscode#1141]
### Summary
There is no code completion for aura components (cmp)

### Steps To Reproduce:

1. sfdx force:lightning:component:create -n mylightningcomp
2. Open mylightningcomp.cmp file
3. Start typing <lightning: then CTRL+SPACE

### Expected result
vs Code should suggest lightning elements

### Actual result
Nothing is suggested :(

### Additional information
In the javascript controllers, LWC elements are suggested. 
And in apex classes, language suggestions are working. 
But I cannot for the life of me get html suggestions working. 

**VS Code Version**:
1.31.1

**SFDX CLI Version**:
6.56

**OS and version**:
Windows 7 SP1 (not fun)

**Comment by lcampos** [comment]
Thanks for opening this @dansad. Yes, the current language server capabilities for lightning are limited but we're working on enhancing that experience. @ntotten can provide more details on it.

**Comment by dansad** [comment]
Thanks @lcampos!
Just to clarify, I cannot get any of the completions working in .cmp html - not even slds or aura components, or anything else that is supposed to be provided by [salesforce.salesforcedx-vscode-lightning](https://marketplace.visualstudio.com/items?itemName=salesforce.salesforcedx-vscode-lightning)

**Comment by lcampos** [comment]
@dansad that's strange. I tested in windows and mac and do see those autocompletions working using version 45.5.0 of the extensions. If you have other (non-salesforce) extensions installed, could you deactivate them, reload vscode and see if that makes the autocompletions work ? I'm thinking there might be another extension causing a conflict.

If that's not the case, could you open VS Code's Dev Tools (Help > Toggle Developer Tools) and see if there are any errors under the console tab ?

**Comment by dansad** [comment]
Strange indeed! I have tried vanilla setups from scratch, and still cannot get going. Just to be sure, I uninstalled all my other extensions, no joy. 

The dev console has several errors `Cannot find en-us, defaulting to en`. That error also shows up in my SLDS Language Server output panel. (It's the only thing in there)

Dev console also informs me that `Deleted environment variables SFDX_DEFAULTUSERNAME and SFDX_INSTANCE_URL` - not sure if that's related. 

Also: ESLint seems to complain when i'm working on JS files:
```[Info  - 10:35:43 AM] ESLint server is running.
[Info  - 10:35:50 AM] ESLint library loaded from: c:\Users\sadavd1\.vscode\extensions\salesforce.salesforcedx-vscode-lwc-45.5.0\node_modules\eslint\lib\api.js
[Warn  - 10:35:50 AM] 
No ESLint configuration (e.g .eslintrc) found for file: PATH_TO_MY_JS_CONTROLLER
File will not be validated. Consider running 'eslint --init' in the workspace folder igm-dces
Alternatively you can disable ESLint by executing the 'Disable ESLint' command.```

Is any of this helpful? 

ps. thanks for looking at this with me. It's been a long hard road working without language support :smiley:

**Comment by Neezen** [comment]
Hi everyone - just joining this conversation as I am having similar issues. To confirm is it correct to see that <lightning:... would not yet autocomplete as well? For example <lightning:attribute does not show any auto completion of code. Curious whether this is expected behaviour or whether my setup is not working. Thanks!

**Comment by adamondrejka** [comment]
Hi. I have same problem, no suggestions and auto completions for aura components in .cmp files, disabling other extensions does not help. Running Visual Studio Code 1.32.3, SalesForce Extension Pack 45.7.0 on macOS Mojave 10.14. No errors in developer tools.

**Comment by steals** [comment]
Autocomplete for Aura components don't work for me either. Here is the stacktrace from Aura Language Server Output panel:

> [Info  - 4:12:18 PM] Starting *AURA* language server at <my_workspace_folder>
Loading support library: browser
Loading support library: ecmascript
[Info  - 4:12:20 PM] ... language server started in 817.43 ms
[Info  - 4:12:32 PM] aura onDidChangeWatchedFiles...
loadCustomTagsFromFiles: processed 1 files in 2275.15 ms
[Info  - 4:12:40 PM] Indexed 90 files in 19595.27 ms ms
TypeError: Cannot read property 'charCodeAt' of undefined
    at isInteger (/Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/infer.js:547:18)
    at exports.Obj.hasProp (/Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/infer.js:581:11)
    at findExprType (/Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/tern.js:918:36)
    at findTypeAtExpr (/Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/tern.js:927:16)
    at Object.findTypeAt [as run] (/Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/tern.js:951:22)
    at run (/Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/tern.js:275:59)
    at /Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/tern.js:278:16
    at Object.exports.withContext (/Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/infer.js:883:18)
    at /Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/tern.js:274:13
    at analyzeAll (/Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/tern.js:404:15)
[Error - 4:13:09 PM] TypeError: Cannot read property 'charCodeAt' of undefined
    at isInteger (/Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/infer.js:547:18)
    at exports.Obj.hasProp (/Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/infer.js:581:11)
    at findExprType (/Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/tern.js:918:36)
    at findTypeAtExpr (/Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/tern.js:927:16)
    at Object.findTypeAt [as run] (/Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/tern.js:951:22)
    at run (/Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/tern.js:275:59)
    at /Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/tern.js:278:16
    at Object.exports.withContext (/Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/infer.js:883:18)
    at /Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/tern.js:274:13
    at analyzeAll (/Users/steals/.vscode/extensions/salesforce.salesforcedx-vscode-lightning-45.8.0/node_modules/aura-language-server/lib/tern/lib/tern.js:404:15)

**Comment by nrkruk** [comment]
This feature was only just released as part of 45.8.0 (last Thursday). If you update to the latest plugin version, you will receive tag completions in .cmp files.

**Comment by ntotten** [comment]
As Nick mentioned, this feature has been released. I am closing this issue, but please reopen or create a new one if you have issues.
